### ISSUE TITLE
Class namespace dropped in a .cpp-file for the templated method of a non-templated class

### ISSUE THREAD

**Issue by ppietrasa** [specs-feup/clava#28]
Given the following header:
```
// ==> TemplatedMethod.h <==
class AKlass {

   void a_regular_method();

   template <class T>
   void a_templated_method(T val);
};
```

And the corresponding .cpp file:

```
// ==> TemplatedMethod.cpp <==
#include <TemplatedMethod.h>


void AKlass::a_regular_method()
{
}

template <class T>
void AKlass::a_templated_method(T val)
{
}
```

and a simple "touch me LARA":
```
aspectdef TouchMeAspect
  select vardecl end
  apply
    setName('a_' + $vardecl.name);
  end
end
```

the .cpp file transpiles to:

```
// ==> woven/TemplatedMethod.cpp <==
#include <TemplatedMethod.h>
void AKlass::a_regular_method() {
}

template <class T>
void a_templated_method(T a_val) {
}
```

Note, that in the case of `a_templated_method` the class namespace prefix is dropped, while the `a_regular_method` is fine. The header was also fine, hence not included here.

**Comment by joaobispo** [comment]
By mistake, the fix to this issue was tagged with issue #29, instead of #28. This is the [commit](https://github.com/specs-feup/clava/commit/f966ececb00cb1e84ae432ba0821119e7eed4977) that fixes this issue.

**Comment by joaobispo** [comment]
I'm closing the issue since it has been solved.
